### ISSUE TITLE
Narrator does not read Tree View description

### ISSUE THREAD

**Issue by wade0016** [microsoft/vscode#239376]
Type: <b>Bug</b>

Only the title of a tree view is being read by Narrator. The accessibility testers for the Microsoft Fabric extension filed a bug noting that the description is not read. Similar behavior exists for the Open Editors tree view when there is an unsaved file.

This could also be a side-effect of [Issue 239375](https://github.com/microsoft/vscode/issues/239375)

VS Code version: Code 1.96.4 (cd4ee3b1c348a13bafd8f9ad8060705f6d4b9cba, 2025-01-16T00:16:19.038Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) Ultra 7 165U (14 x 2688)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|63.58GB (30.78GB free)|
|Process Argv|--open-url --crash-reporter-id 8a56e54a-32a0-44cb-a14d-672ca35d9cad -- vscode://fabric.vscode-fabric-functions/?workspaceId=4ce7129c-5378-4709-84e1-1603692cfaa2&&artifactId=ebfd756b-8e44-4028-86aa-78a924acfede&&Environment=DAILY|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (23)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-fabric|fab|0.13.2
vscode-fabric-functions|fab|0.13.5
vscode-azurefunctions|ms-|1.16.1
vscode-azureresourcegroups|ms-|0.10.4
csdevkit|ms-|1.16.4
csharp|ms-|2.61.28
vscode-dotnet-runtime|ms-|2.2.6
data-workspace-vscode|ms-|0.5.0
mssql|ms-|1.28.0
sql-bindings-vscode|ms-|0.4.0
sql-database-projects-vscode|ms-|1.4.5
debugpy|ms-|2024.14.0
python|ms-|2024.22.2
vscode-pylance|ms-|2024.12.1
datawrangler|ms-|1.16.0
jupyter|ms-|2025.1.2024121801
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.21
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
command-executor|mwa|0.0.1
synapse|Syn|1.11.0
tree-view-changer|und|0.0.1

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
14424-chatspm:31229291
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dwnewjupytercf:31046870
nativerepl2:31139839
pythonrstrctxt:31112756
nativeloc1:31192215
iacca1:31171482
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624
dwoutputs:31217127
hdaa2157:31222309
copilot_t_ci:31222730

```

</details>

**Comment by alexr00** [comment]
Looks like you're refering to the view badge:

![Image](https://github.com/user-attachments/assets/34cfba5f-b62c-419e-9cef-3dc60b519f58)

**Comment by alexr00** [comment]
I can understand what you're referring to with the "Open Editors" view, but I don't understand for other tree views. Could you share a screen shot?

**Comment by TylerLeonhardt** [comment]
Maybe this is expected... but just in case I reopen... I don't get the description read unless the header is focused.

<img width="343" alt="Image" src="https://github.com/user-attachments/assets/f3e00f56-6cdd-4be5-a20f-762b084f9f41" />

if the view contents is focused it only says "Test View tree view"

<img width="368" alt="Image" src="https://github.com/user-attachments/assets/4833f6cd-461b-43b1-a00d-fa700872841c" />

This is with NVDA on Windows

All I did was grabbed our tree view sample and added

```
view.description = 'Testing';
```

after

https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/testView.ts#L6

**Comment by alexr00** [comment]
I only added it for the view header. @meganrogge, should it also read the description when the tree is focused?

**Comment by alexr00** [comment]
I'm going to keep this as is. I don't think we need to read the description when the tree is focused.

**Comment by meganrogge** [comment]
I think it's fine for it to not read the description when the tree is focused because a user first has to focus the header, which has that.
